src/backend/XmlHttpRequest.ts

```diff
@@ -390,16 +415,6 @@ export default class XmlHttpRequest extends BaseFileSystem implements FileSystem
   }
 
   /**
-   * Asynchronously download the given file.
-   */
-  private _requestFileAsync(p: string, type: 'buffer', cb: BFSCallback<Buffer>): void;
-  private _requestFileAsync(p: string, type: 'json', cb: BFSCallback<any>): void;
-  private _requestFileAsync(p: string, type: string, cb: BFSCallback<any>): void;
-  private _requestFileAsync(p: string, type: string, cb: BFSCallback<any>): void {
-    asyncDownloadFile(this.getXhrPath(p), type, cb);
-  }
-
-  /**
    * Synchronously download the given file.
    */
   private _requestFileSync(p: string, type: 'buffer'): Buffer;
```

Comment:
Mmm. You're right, I think that's a good idea. I thought that sync filesystem functions would be using the `supportsSynch` property to check whether the fs supported sync functions before calling them? If not then yes we should definitely stub these for the no-XHR case.